app/models/template.rb

```diff
@@ -99,6 +101,16 @@ def import_without_save(text, options = {})
     self
   end
 
+  def self.valid_params?(name, metadata)
+    if (!metadata["kind"].nil? && metadata["kind"] != "snippet") && (metadata["snippet"] == true)
+      raise Foreman::Exception.new(N_('When snippet is set as true, template kind shouldn\'t be present'))
+    end
+    unless name.eql?(metadata["name"])
```

Comment:
Yes, I agree with you. It is a bit problematic when it forces the metadata. 
When I talked with @lzap about it, he suggested the name validation. 
It is supposed to ensure that the name corresponds to the metadata if exists. maybe that is the change I should do. 
@lzap Do you still think it is a good idea? :)

Reply:
I believe we do have that validation sort of, the template is looked for by name. Could we implement the `template_kind` thing in this PR and possibly do the name validation separetly if we still feel it is important? It would make use move forward and simplify this PR

Reply:
This is absolutely not important, I just was just throwing few ideas what we could possibly do without thinking it through.